spec/git-repository-async-spec.js

```diff
@@ -0,0 +1,457 @@
+'use babel'
+
+const fs = require('fs-plus')
+const path = require('path')
+const temp = require('temp')
+const Git = require('nodegit')
+
+const GitRepositoryAsync = require('../src/git-repository-async')
+const Project = require('../src/project')
+
+temp.track()
+
+function openFixture(fixture) {
+  return GitRepositoryAsync.open(path.join(__dirname, 'fixtures', 'git', fixture))
+}
+
+function copyRepository() {
+  let workingDirPath = temp.mkdirSync('atom-working-dir')
+  fs.copySync(path.join(__dirname, 'fixtures', 'git', 'working-dir'), workingDirPath)
+  fs.renameSync(path.join(workingDirPath, 'git.git'), path.join(workingDirPath, '.git'))
+  return fs.realpathSync(workingDirPath)
+}
+
+function asyncIt(name, fn) {
+  it(name, () => {
+    waitsForPromise(fn)
+  })
+}
+
+function fasyncIt(name, fn) {
+  fit(name, () => {
+    waitsForPromise(fn)
+  })
+}
+
+function xasyncIt(name, fn) {
+  xit(name, () => {
+    waitsForPromise(fn)
+  })
+}
```

Comment:
@nathansobo ended up adding functions like these in [async-spec-helpers.coffee](https://github.com/atom/atom/blob/master/spec/async-spec-helpers.coffee). They detect if a promise is returned from the `it` function, so you can just shadow `it` and friends.